packages/transformers/babel/src/babel7.js

```diff
@@ -50,6 +57,45 @@ export default async function babel7(
     );
   } else {
     res = await babel.transformAsync(await asset.getCode(), config);
+    if (res.ast) {
+      let map = await asset.getMap();
+      if (map) {
+        // remap ast to original mappings
+        // This improves sourcemap accuracy and fixes sourcemaps when scope-hoisting
+        traverse(res.ast.program, {
```

Comment:
Not entirely sure if we can speed this up with a custom traverser? Don't think babylon-walk allows this.

Reply:
You could change the `simple` traversal in babylon walk to also accept a callback instead of an object.

Reply:
Thanks, will look into that